components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/UpdateAdvancedConfigurations.java

```diff
@@ -61,8 +61,14 @@ private void updateCertificate(Certificate certificate, ServiceProvider serviceP
         if (certificate != null) {
             if (certificate.getType() == Certificate.TypeEnum.PEM) {
                 setIfNotNull(certificate.getValue(), serviceProvider::setCertificateContent);
+                if (certificate.getValue() != null) {
+                    serviceProvider.setJwksUri(null);
```

Comment:
@nipunthilakshan @mefarazath In the data model we allow having both the JWKS URL and the PEM certificate, then from the service layers, and UI we only allow to have either one of them. I feel this makes the code complicated thus leads to these types of issues. Shall we keep one attribute in the object model to store content and another one for the type to model the behaviour of allowing only one certificate type? An example would be,

`serviceProvider.getCertificateContent()` & `serviceProvider.getCertificateType()`

Here content is a string filed where type can be either PEM or JWKS. When the type is PEM -> content is a URL in String form.

Is this doable with the use case requirements?

Reply:
+1

Before we do the proposed would be better to announce the change in a public mail to see if there are use cases to support both.